osu.Game.Rulesets.Taiko/Difficulty/Preprocessing/TaikoDifficultyHitObject.cs

```diff
@@ -68,33 +67,31 @@ public class TaikoDifficultyHitObject : DifficultyHitObject
                                         List<TaikoDifficultyHitObject> noteObjects, int index)
             : base(hitObject, lastObject, clockRate, objects, index)
         {
-            // Create the Colour object, its properties should be filled in by TaikoDifficultyPreprocessor
-            Colour = new TaikoDifficultyHitObjectColour();
-
-            var currentHit = hitObject as Hit;
             noteDifficultyHitObjects = noteObjects;
 
+            // Create the Colour object, its properties should be filled in by TaikoDifficultyPreprocessor
+            Colour = new TaikoDifficultyHitObjectColour();
             Rhythm = getClosestRhythm(lastObject, lastLastObject, clockRate);
-            HitType? hitType = currentHit?.Type;
 
-            if (hitType == HitType.Centre)
+            switch ((hitObject as Hit)?.Type)
             {
-                MonoIndex = centreHitObjects.Count;
-                centreHitObjects.Add(this);
-                monoDifficultyHitObjects = centreHitObjects;
+                case HitType.Centre:
+                    MonoIndex = centreHitObjects.Count;
+                    centreHitObjects.Add(this);
+                    monoDifficultyHitObjects = centreHitObjects;
+                    break;
+
+                case HitType.Rim:
+                    MonoIndex = rimHitObjects.Count;
+                    rimHitObjects.Add(this);
+                    monoDifficultyHitObjects = rimHitObjects;
+                    break;
+
+                default:
+                    NoteIndex = noteObjects.Count;
+                    noteObjects.Add(this);
+                    break;
             }
-            else if (hitType == HitType.Rim)
-            {
-                MonoIndex = rimHitObjects.Count;
-                rimHitObjects.Add(this);
-                monoDifficultyHitObjects = rimHitObjects;
-            }
-
-            // Need to be done after HitType is set.
-            if (hitType == null) return;
-
-            NoteIndex = noteObjects.Count;
-            noteObjects.Add(this);
```

Comment:
Doesn't this change break the previous behavior of also adding the hitobject to the `noteObjects` array regardless of type?
Before this change, notes with `HitType.Rim` and `HitType.Centre` were also added, now they wont be.

Reply:
Yes, it broke tests. I fixed it.